docs/book/payments.rst

```diff
@@ -4,112 +4,136 @@
+.. code-block:: php
 
-* new (initial)
-* unknown
-* pending
-* processing
-* completed
-* failed
-* cancelled
-* void
-* refunded
+    $payment = $this->container->get('sylius.factory.payment')->createNew();
 
-Of course, you can define your own states and transitions to create a workflow, that perfectly matches your business. Full configuration can be seen `here <https://github.com/Sylius/Sylius/blob/master/src/Sylius/Bundle/PaymentBundle/Resources/config/app/state_machine.yml>`_.
+    $payment->setOrder($order);
+    $payment->setCurrencyCode('USD');
 
-Changes to payment happen mostly through applying appropriate transitions.
+    $this->container->get('sylius.repository.payment')->add($payment);
 
 Payment Methods
 ---------------
 
-A [PaymentMethod] represents a way that your customer pays during the checkout process. It holds a reference to specific *gateway* with custom configuration.
-You can have different payment methods using the same gateway, like PayPal or Stripe. Default model of payment method contains the following fields:
+A [PaymentMethod] represents a way that your customer pays during the checkout process. It holds a reference to a specific *gateway* with custom configuration.
+You can have different payment methods using the same gateway, like PayPal or Stripe.
+
+How to create a PaymentMethod programmatically?
+'''''''''''''''''''''''''''''''''''''''''''''''
+
+As usually use a factory to create a new PaymentMethod, give it a unique code.
+The available gateways of your system are available under the ``sylius.payment_gateways`` parameter.
 
-* name
-* description
-* enabled
-* gateway
-* configuration
-* environment
-* feeCalculator
-* feeCalculatorConfiguration
-* createdAt
-* updatedAt
+.. code-block:: php
 
-Gateway and Configurations
---------------------------
+    $paymentMethod = $this->container->get('sylius.factory.payment_method')->createNew();
+    $paymentMethod->setCode('ALFA1');
 
-...
+    $gateways = $this->container->getParameter('sylius.payment_gateways');
 
-Payment Processing
-------------------
+    $paymentMethod->setGateway($gateways['offline']);
 
-...
+    $this->container->get('sylius.repository.payment_method')->add($paymentMethod);
 
-Supported Gateways
-------------------
+In order to have your new payment method available in the checkout remember to **add it to your desired channels**:
 
-...
+.. code-block:: php
 
-Final Thoughts
---------------
+    $channel->addPaymentMethod($paymentMethod);
 
-...
+Payment Gateway and Configurations
+----------------------------------
 
-Learn more
-----------
+In order to add a new gateway configure it in the ``app/config_dev.yml`` file of your project in such a way:
 
-* ...
+.. code-block:: yaml
+
+    payum:
+        gateways:
+            paypal_express_checkout:
+                factory: "paypal_express_checkout"
+                payum.http_client: "@sylius.payum.http_client"
```

Comment:
I'm not sure if I've got enough knowledge to describe that. \cc @Arminek - will u help me?